core-web/libs/sdk/experiments/src/lib/parser/parser.ts

```diff
@@ -63,3 +63,122 @@ const verifyRegex = (regexToCheck: string | null, href: string): boolean => {
         return false;
     }
 };
+
+/**
+ * This function merges newly fetched data with the data stored from IndexedDB, preparing it for re-storage in IndexedDB.
+ *
+ * @param { AssignedExperiments | null } fetchExperiments - The experiment data fetched from the API.
+ * @param { AssignedExperiments | null } storedExperiments - The experiment data currently stored in IndexedDB.
+ *
+ * @returns { AssignedExperiments } - The parsed experiment data ready for storing.
+ *
+ * Following cases are handled -
+ * 1) When new Data is received without Old data. This is assumed to be the first time data is received.
+ * 2) When only old data is received, implying that the timestamp hasn't expired and the data is available in IndexedDB.
+ *    The data is verified and any expired experiment is removed before being assigned to dataToStorage.
+ * 3) When both old data and new data is present. This implies that the record existed in IndexedDB but was fetched because the flag had expired.
+ *    Additional operations are performed to add expiry time to experiments, merging all experiments, and storing them.
+ *
+ * There could be scenarios where none of these conditions are met, in that case, dataToStorage will be the default empty object.
+ */
+export const parseData = (
+    fetchExperiments: AssignedExperiments | null,
+    storedExperiments: AssignedExperiments | null
+): AssignedExperiments => {
+    let dataToStorage: AssignedExperiments = {} as AssignedExperiments;
+
+    if (fetchExperiments && !storedExperiments) {
+        // TODO: Use fetchExperiment instead fetchExperimentsNoNoneExperimentID when the endpoint dont retrieve NONE experiment
+        // https://github.com/dotCMS/core/issues/27905
+        const fetchExperimentsNoNoneExperimentID: AssignedExperiments = {
```

Comment:
This name is kind of confusing and long, can we find another one?

Reply:
It is more a think internal while [#27905](https://github.com/dotCMS/core/issues/27905) will be added to the endpoint. After that, I will remove from 93-98.